agent/credentials/manager.go

```diff
@@ -57,6 +57,15 @@ type IAMRoleCredentials struct {
 type TaskIAMRoleCredentials struct {
 	ARN                string
 	IAMRoleCredentials IAMRoleCredentials
+	lock               sync.RWMutex
```

Comment:
Who's acquiring this lock for writes?

Reply:
Currently no one is using writing lock, as it was not modified anywhere after created.  I think it would still be good to have a lock as we have a reader here, where we probably will have a writer in the future. I can remove this lock if you are strongly opposite of this.